src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,699 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import {PCVMover} from "@voltprotocol/pcv/PCVMover.sol";
+import {Constants} from "@voltprotocol/Constants.sol";
+import {CoreRefV2} from "@voltprotocol/refs/CoreRefV2.sol";
+import {IPCVOracle} from "@voltprotocol/oracle/IPCVOracle.sol";
+import {IPCVDeposit} from "@voltprotocol/pcv/IPCVDeposit.sol";
+import {IPCVDepositV2} from "@voltprotocol/pcv/IPCVDepositV2.sol";
+import {IMarketGovernance} from "@voltprotocol/vcon/IMarketGovernance.sol";
+import {DeviationWeiGranularity} from "@voltprotocol/utils/DeviationWeiGranularity.sol";
+
+import {console} from "@forge-std/console.sol";
+
+/// @notice this contract requires the PCV Controller and Locker role
+///
+/// Core formula for market governance rewards:
+///     Profit Per VCON = Profit Per VCON + ∆Cumulative Profits (Dollars) * VCON:Dollar  / VCON Staked
+///
+/// If an account has an unrealized loss on a venue, they cannot do any other action on that venue
+/// until they have called the function realizeLosses and marked down the amount of VCON they have staked
+/// on that venue. Once the loss has been marked down, they can proceed with other actions.
+///
+/// The VCON:Dollar ratio is the same for both profits and losses. If a venue has a VCON:Dollar ratio of 5:1
+/// and the venue gains $5 in profits, then 25 VCON will be distributed across all VCON stakers in that venue.
+/// If that same venue losses $5, then a loss of 25 VCON will be distributed across all VCON stakers in that venue.
```

Comment:
@dev this contract assumes it is already topped up with the VCON necessary to pay rewards. A dripper will keep this contract funded at a steady pace.